extension/deployment/src/test/java/io/quarkiverse/temporal/deployment/WorkflowStubMultipleInjectionTest.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkiverse.temporal.deployment;
+
+import static io.quarkiverse.temporal.Constants.DEFAULT_WORKER_NAME;
+
+import jakarta.enterprise.inject.Produces;
+import jakarta.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkiverse.temporal.WorkflowStub;
+import io.quarkiverse.temporal.deployment.stub.NamedSimpleWorkflowImpl;
+import io.quarkiverse.temporal.deployment.stub.SimpleWorkflow;
+import io.quarkiverse.temporal.deployment.stub.UnnamedSimpleWorkflowImpl;
+import io.quarkus.test.QuarkusUnitTest;
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.client.WorkflowClient;
+import io.temporal.testing.TestWorkflowEnvironment;
+
+public class WorkflowStubMultipleInjectionTest {
+
+    public static TestWorkflowEnvironment testWorkflowEnvironment = TestWorkflowEnvironment.newInstance();
+
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(SimpleWorkflow.class)
+                    .addClass(UnnamedSimpleWorkflowImpl.class)
+                    .addClass(NamedSimpleWorkflowImpl.class)
+                    .addAsResource(
+                            new StringAsset("\n"),
+                            "application.properties"));
+
+    @Inject
+    WorkflowClient workflowClient;
```

Comment:
@melloware another weirdness: if I remove this injection, the test fail with:
```
Caused by: java.lang.IllegalStateException: No bean type found for: PRODUCER METHOD bean [types=[io.temporal.client.WorkflowClient, java.lang.Object], qualifiers=[@Default, @Any], target=io.temporal.client.WorkflowClient client(), declaringBean=io.quarkiverse.temporal.deployment.WorkflowStubMultipleInjectionTest$Producers]
```

Reply:
If you remove that injection then how will it get a WorkflowClient?

Reply:
OK yeah this is bad then.  If you don't need the WorkflowClient here it shouldn't have to be injected so we have to somehow provide `@Alternative` bean or something that allows your code to work without this injection.

Reply:
Or shouldn't a default one always be `produced` by this extension?